src/Middleware/ImplicitHeadMiddleware.php

```diff
@@ -92,7 +96,7 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         }
 
         if (! $route->allowsMethod(RequestMethod::METHOD_GET)) {
-            return $this->response;
+            return ($this->responseFactory)(IMPLICIT_HEAD_MIDDLEWARE_RESPONSE);
```

Comment:
What do you think? Do we need to pass it to the factory or it is completely useless?
I think it could give us more flexibility, in the factory when we know what response we should create.

Maybe we can pass just middleware name instead of constant and then we can get rid of constants at all?

Reply:
This would be incompatible with the proposed PSR-17. The proposed `ResponseFactoryInterface::createResponse()` accepts an optional argument, but it's the response status.

Any response factory we create should simply return a new, general response. Consumers can then manipulate that instance using the various `with*()` methods in order to produce the final response to return.

Reply:
Don't pass any arguments... which allows us to remove all constants.